src/main/java/com/synopsys/integration/detect/lifecycle/autonomous/AutonomousManager.java

```diff
@@ -31,16 +37,23 @@ public class AutonomousManager {
     private boolean autonomousScanEnabled;
     private final DetectConfigurationFactory detectConfigurationFactory;
     private final DetectPropertyConfiguration detectConfiguration;
+    private SortedMap<String, String> userProvidedProperties = new TreeMap<>();
+    private SortedMap<String, String> globalProperties = new TreeMap<>();
+    private SortedMap<String, String> detectorSharedProperties = new TreeMap<>();
+    private static final List<String> propertiesNotAutonomous = Arrays.asList("blackduck.api.token", "detect.diagnostic", "detect.blackduck.scan.mode", "detect.source.path", "detect.tools");
```

Comment:
Curious why detect.diagnostic is included in this list?

Reply:
As detect.diagnostic can also be used as a flag and currently we are not including flags in the scan settings file, I am not including it in the file right now.

Reply:
Can you file a tech debt ticket later to track?